### PR TITLE
ServiceBus Topic Authorization Rules

### PR DESCRIPTION
**Pull request by scottrangerio**
Hey all o/

This PR addresses https://github.com/terraform-providers/terraform-provider-azurerm/issues/735

Tests pass and output can be seen here https://gist.github.com/scottrangerio/501719dd8b5cd7f2915a5350690cd237

Also tested manually using the following .tf file
https://gist.github.com/scottrangerio/37fa55911682cdd1e32f8905c9e68c33

This is a near complete copy of the eventhub_authorization_rule resource, just tweaked for servicebus topics

Documentation still needs doing, but I'm happy to tackle that at some point next week when I get a bit more time!

### CODE REVIEW

File: azurerm/resource_arm_servicebus_topic_authorization_rule.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+	resGroup := id.ResourceGroup
+	namespaceName := id.Path["namespaces"]
+	topicName := id.Path["topics"]
+	name := id.Path["authorizationRules"]
+
+	resp, err := client.DeleteAuthorizationRule(ctx, resGroup, namespaceName, topicName, name)
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Error issuing Azure ARM delete request of ServiceBus Topic Authorization Rule '%s': %+v", name, err)
+	}
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_servicebus_topic_authorization_rule.go]: checking the SDK here, the method above can return either a 200 or a 204 - can we change this to:

```
if err != nil {
  return fmt.Errorf("Error issuing Azure ARM delete request of ServiceBus Topic Authorization Rule %q (Resource Group %q): %+v", name, resGroup, err)
}
```

File: website/docs/r/servicebus_topic_authorization_rule.html.markdown [hunk 2]
```diff
@@ -0,0 +1,94 @@
+---
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_servicebus_topic_authorization_rule"
+sidebar_current: "docs-azurerm-resource-servicebus-topic-authorization-rule"
+description: |-
+  Creates a new ServiceBus Topic authorization Rule within a ServiceBus Topic.
+---
+
+# azurerm\_servicebus\_topic\_authorization\_rule
```
Comment (tombuildsstuff) [inline website/docs/r/servicebus_topic_authorization_rule.html.markdown]: can we remove the `\` here? they're not needed

File: website/docs/r/servicebus_topic_authorization_rule.html.markdown [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+resource "azurerm_servicebus_topic" "test" {
+  name                = "testTopic"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  namespace_name      = "${azurerm_servicebus_namespace.test.name}"
+}
+
+resource "azurerm_servicebus_topic_authorization_rule" "test" {
+  name                = "navi"
```
Comment (tombuildsstuff) [inline website/docs/r/servicebus_topic_authorization_rule.html.markdown]: could we make this `examplerule`?

**Comment by scottrangerio** [comment]
Hey @tombuildsstuff 

No worries about the delay, it's nice to see the amount of activity this repo is gaining :) I've made said changes and rebased on top of master (first time using rebase so I hope I've done it right?). Any other issues just let me know and I'll try my best to resolve them.

Cheers!

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2018-02-26 at 18 51 14

**Comment by tombuildsstuff** [comment]
:wave: hey @scottrangerio   

Just to let you know that support for this has just been released in v1.2.0 of the AzureRM Provider - full details of what's included are available here: https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.2.0/CHANGELOG.md#120-march-02-2018

Thanks!

**Pull request merged**